docs/source/features/version_control.rst

```diff
@@ -2,25 +2,180 @@
+
+   ACCESS_KEY = "Accesskey-*****"
+   gas = GAS(ACCESS_KEY)
+   commits = dataset_client.list_commits()
+
+   FIRST_COMMIT_ID = "28f40062271d480ea9be39e7d1beb64a"
+   version_control_demo = Dataset("VersionControlDemo", gas, revision=FIRST_COMMIT_ID)
+   train_segment = version_control_demo["train"]
+   print(f"data number: {len(train_segment)}.")
+   # data number: 4.
+
+There are 4 data in the train segment.
+
+Checkout to the second commit and check the data amount.
+
+.. code:: python
+   
+   SECOND_COMMIT_ID = "f7d0c00cc1824097b10f7846353697cc"
+   version_control_demo = Dataset("VersionControlDemo", gas, revision=SECOND_COMMIT_ID)
+   train_segment = version_control_demo["train"]
+   print(f"data number: {len(train_segment)}.")
+   # data number: 8.
+
+There are 8 data in the train segment.
+
+See :ref:`version_control/draft_and_commit:Draft and Commit` for more details about commit.
+
+*******
+ Draft
+*******
+
+So how to create a dataset with multiple commits?
+Actually, each commit comes from a :ref:`reference/glossary:draft`.
+A draft is a concept that represents a **writable** workspace, which is built upon a commit.
+Typical steps to create a new commit:
+
+- Create a draft.
+- Do the modifications/update.
+- Commit the draft.
+
+:numref:`Figure. %s <draft-demonstration>` demonstrates the relations between a draft and commits.
+
+.. _draft-demonstration:
+
+.. figure:: ../images/draft.png
+   :scale: 30 %
    :align: center
 
-   The relations between different versions of a dataset.
+   The relations between a draft and commits.
+
+The following code block creates a draft and adds a new segment to the "VersionControlDemo" dataset.
+
+.. code:: python
+
+   import os
+   from tensorbay.dataset import Segment
+
+   TEST_IMAGES_PATH = "path/to/test_images"
+
+   dataset_client = gas.get_dataset("VersionControlDemo")
+   dataset_client.create_draft("draft-1")
+
+   test_segment =Segment("test")
+
+   for image_name in os.listdir(TEST_IMAGES_PATH):
+       data = Data(os.path.join(TEST_IMAGES_PATH, image_name))
+       test_segment.append(data)
+
+   dataset_client.upload_segment(test_segment, jobs=8)
+   dataset_client.commit("add test segment")
 
+See :ref:`version_control/draft_and_commit:Draft and Commit` for more details about draft.
+
+*****
+ Tag
+*****
+
+For the convenience of marking major commits and switching between different commits,
+TensorBay provides the :ref:`reference/glossary:tag` concept.
+The typical usage of tag is to mark release revisions.
+
+The tag "v1.0.0" in :numref:`Fig. %s <commit-demonstration>` is added by
+
+.. code:: python
+
+   dataset_client.create_tag("v1.0.0", revision=SECOND_COMMIT_ID)
+
+See :ref:`version_control/tag:Tag` for more details about tag.
+
+********
+ Branch
+********
+
+Sometimes, users may need to create drafts upon an early (not the latest) commit.
+For example, in an algorithm team, each team member may do modifications/update to every existed commit.
```

Comment:
Is `every` appropriate?
`do modifications/update based on different versions of the dataset`?